olive/passes/onnx/gptq.py

```diff
@@ -0,0 +1,149 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+import logging
+from typing import Any, Dict
+
+import torch
+
+from olive.hardware.accelerator import AcceleratorSpec
+from olive.model import PyTorchModelHandler
+from olive.model.handler.onnx import ONNXModelHandler
+from olive.passes import Pass
+from olive.passes.onnx.common import get_external_data_config
+from olive.passes.onnx.gptq_utils import QuantLinearORT
+from olive.passes.pass_config import PassConfigParam
+
+logger = logging.getLogger(__name__)
+
+
+class GptqQuantizer(Pass):
+    """GPTQ quantization using Hugging Face Optimum and export model with onnxruntime optimized kernel."""
+
+    @staticmethod
+    def _default_config(accelerator_spec: AcceleratorSpec) -> Dict[str, PassConfigParam]:
+        config = {
+            "nsamples": PassConfigParam(
+                type_=int,
+                default_value=128,
+                description="number of samples in calibration dataset to apply quantization. Default value is 128",
+            ),
+            "bits": PassConfigParam(
+                type_=int,
+                default_value=4,
+                description="quantization bits. Default value is 4",
+            ),
+            "dataset": PassConfigParam(
+                type_=str,
+                default_value="wikitext2",
+                description="Calibration dataset. Default value is wikitext2",
+            ),
+            "block_name_to_quantize": PassConfigParam(
+                type_=str,
+                default_value=None,
+                description="Block name to quantize. Default value is model.decoder.layers.",
```

Comment:
What is the default value here?

Reply:
None is acceptable to pass in AutoModel.from_pretrained. I haven't use optimum quantizer directly for it requires this parameter.

Reply:
I mainly worry about the default values are different between actual default value and description. We should update either one to make them aligned.